staging/src/k8s.io/client-go/testing/fixture_test.go

```diff
@@ -274,3 +275,26 @@ func TestPatchWithMissingObject(t *testing.T) {
 	assert.Nil(t, node)
 	assert.EqualError(t, err, `nodes "node-1" not found`)
 }
+
+func TestGetWithExactMatch(t *testing.T) {
+	scheme := runtime.NewScheme()
+	codecs := serializer.NewCodecFactory(scheme)
+	o := NewObjectTracker(scheme, codecs.UniversalDecoder())
+
+	nodesResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "node"}
+	node := getArbitraryResource(nodesResource, "node", "")
+	gvks, _, err := scheme.ObjectKinds(node)
+	assert.Nil(t, err)
+
+	pluralGvk, _ := meta.UnsafeGuessKindToResource(gvks[0])
+
+	assert.Nil(t, o.Add(node))
+
+	_, err = o.Get(pluralGvk, "", "node")
+	assert.Nil(t, err)
+
+	_, err = o.Get(pluralGvk, "ns", "node")
+	assert.NotNil(t, err)
+	errNotFound := errors.NewNotFound(pluralGvk.GroupResource(), "node")
+	assert.EqualError(t, err, errNotFound.Error())
+}
```

Comment:
Can you verify that an error is returned if o.Get is called with empty namespace, but the actual object has a namespace?

Reply:
Done